### PR TITLE
ARROW-6167: [R] macOS binary R packages on CRAN don't have arrow_available

### PR DESCRIPTION
**Pull request by nealrichardson**
This fixes one of the three possibilities described on the ticket, and certainly not the one that affected CRAN. It also adds a warning to the README with workaround instructions.

### CODE REVIEW

File: r/configure [hunk 1]
```diff
@@ -48,41 +48,41 @@ else
...
     PKG_LIBS=${PKGCONFIG_LIBS}
   else
     if [[ "$OSTYPE" == "darwin"* ]]; then
       if [ "$(command -v brew)" ]; then
-        echo "brew is available"
-        BREWDIR=$(brew --prefix)
-
-        if brew ls --versions apache-arrow > /dev/null; then
-          # Install apache-arrow via Homebrew if we don't already have it
+        if [ "$(brew ls --versions apache-arrow)" == "" ]; then
```
Comment (kou) [inline r/configure]: Can we use `${PKG_BREW_NAME}` instead of `apache-arrow` here and other `apache-arrow`s too?

File: r/configure [hunk 2]
```diff
@@ -48,41 +48,41 @@ else
...
-    echo "PKGCONFIG_CFLAGS = \"${PKGCONFIG_CFLAGS}\""
+    # echo "PKGCONFIG_CFLAGS = \"${PKGCONFIG_CFLAGS}\""
 
-    echo "$ pkg-config --libs ${PKG_CONFIG_NAME}"
+    # echo "$ pkg-config --libs ${PKG_CONFIG_NAME}"
     PKGCONFIG_LIBS=`pkg-config --libs --silence-errors ${PKG_CONFIG_NAME}`
-    echo "PKGCONFIG_LIBS = \"${PKGCONFIG_LIBS}\""
-    echo ""
+    # echo "PKGCONFIG_LIBS = \"${PKGCONFIG_LIBS}\""
+    # echo ""
```
Comment (kou) [inline r/configure]: Can we remove these `echo`s instead of commenting out them?

**Review by kou — APPROVED** [review]
+1 with some trivial improvements.

File: r/configure (hunk 1 above)
Comment (nealrichardson) [reply r/configure]: Done, and also removed the commented echos

**Comment by kou** [comment]
Could you resolve conflict for `r/README.Rmd`? Then I'll merge this.

**Comment by nealrichardson** [comment]
Done

**Pull request closed without merging**